ykrt/src/compile/jitc_yk/codegen/x86_64/lsregalloc.rs

```diff
@@ -0,0 +1,1099 @@
+//! A simple linear scan register allocator.
+//!
+//! The "main" interface from the code generator to the register allocator is `get_gp_regs` (and/or
+//! `get_fp_regs`) and [RegConstraint]. For example:
+//!
+//! ```rust,ignore
+//! match binop {
+//!   BinOp::Add => {
+//!     let size = lhs.byte_size(self.m);
+//!     let [lhs_reg, rhs_reg] = self.ra.get_gp_regs(
```

Comment:
Is there a way to give back registers that are used only as temporarys, or is that also future work?

Reply:
There is no concept of "temporary" in the register allocator (yet). When you get registers, they are valid to use for the JIT IR instruction you're generating code for -- and no longer. The register allocator will decide, at the start of the next instruction, whether it wants to keep those registers untouched or reuse them (etc).